### PR TITLE
OCD-3307 - Add 2015 Cures Edition metrics to summary statistics report

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-3307

### CODE REVIEW

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationResultDAO.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertificationResultDAO.java]: Do you need an "and cr.deleted = false" type of statement as part of the "IN" clause? Maybe not but I can't tell just from thinking about it.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/AsynchronousSummaryStatistics.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/AsynchronousSummaryStatistics.java]: For checkstyle purposes could the string "ONC Test Method" be a constant?

**Review by kekey1 — CHANGES_REQUESTED** [review]
My two comments may not be required changes (not sure about the HQL one) so if you think it's all correct how it is I won't hold up the PR for that.. let me know.

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-resources/src/main/resources/jobs.xml
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/jobs.xml]: Is this necessary? I think I'd rather not do it by default on local environments, which is the only place this job runs

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertifiedProductDAO.java
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertifiedProductDAO.java]: `@Log4j2` ?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsEmailJob.java
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsEmailJob.java]: Should this be here?

File: ready-for-integration.sh
Comment (andlar) [inline ready-for-integration.sh]: This should be ignored in the .gitignore file. It is on the staging branch now

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/EditionCriteria.java
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/EditionCriteria.java]: Worth putting these on multiple lines so when we (inevitably) need to change them it's easier to read the line diff?

**Review by andlar — COMMENTED** [review]
There's a PR for OCD-3275 that has a "maybe going to be standard" email signature thing in it. Might be worth checking to see if that should be put in this email as well. I know it'll block this until that one gets merged, but that one is ready to merge now, so probably won't block for long

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/CsvStatistics.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/CsvStatistics.java]: EmailStatistic and EmailStatistics are in the domain package and CSVStatistics is in the job/summarystatistics package. Just based on the class names I would think they should probably all be in the job/summarystatistics package because "Email" is really specific to what the job does (as is CSV). Thoughts on the naming or location of these?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/NonConformityStatisticsSectionCreator.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/NonConformityStatisticsSectionCreator.java]: Just another naming thing for this class and at least one other.. I think we usually write "Nonconformity" as a single word without camel case. At least it is that way in most of the surveillance code. Now I wonder if that is wrong since I sometimes see it written with a dash (Non-conformity)... could we figure out which way is right/which way we want to have it in the code and move towards that?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/StatisticsMassager.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/StatisticsMassager.java]: Can you rename to getNewCertificationBodyStatistic (instead of CertifiedBody)? I think we use CertificationBody more consistently in the code although maybe not everywhere.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsEmailJob.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsEmailJob.java]: I don't see this used anywhere

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertifiedProductDAO.java
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dao/CertifiedProductDAO.java]: FYI I have a PR for OCD-3275 ("Reviewed" status) that also removes this Impl. It would be good to make sure we know what other changes were made to the file in each case so we don't lose them because this will be a conflict for sure (I would think if you pushed this to DEV it has already become an issue there?).

**Review by kekey1 — COMMENTED** [review]
Most of my comments are actually just comments and don't require changes but I want to discuss a couple of them before settling on Approval.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/NonConformityStatisticsSectionCreator.java
Comment (andlar) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/NonConformityStatisticsSectionCreator.java]: I like this suggestion, but think we also need to look at the UI to see what the "business domain" usage has been. A quick check suggests that all of "Nonconformity", "Non-conformity", and "Non-Conformity" are in use (though the middle one only once) and with "Non-Conformity" the slight favorite. Suggest we bring it to the team and get an answer from the PO, then move towards whatever answer that is everywhere we can

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**
